src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/SafeTransactionPreview.tsx

```diff
@@ -119,84 +138,173 @@ interface Props {
   values: FormValues;
 }
 
-const SafeTransactionPreview = ({ colony: { tokens }, values }: Props) => {
-  const transactionDetails = useMemo(
-    () =>
-      values.transactions[0].transactionType === ''
-        ? []
-        : transactionTypeFieldsMap[values.transactions[0].transactionType],
-    [values],
+const SafeTransactionPreview = ({ colony, values }: Props) => {
+  const [transactionTabStatus, setTransactionTabStatus] = useState(
+    Array(values.transactions.length).fill(false),
   );
 
-  const token = useMemo(
-    () =>
-      tokens.find(
-        (item) => item.address === values.transactions[0].tokenAddress,
-      ),
-    [values, tokens],
+  const tokens = useMemo(() => {
+    return values.transactions.map((transaction) => {
+      const token = colony.tokens.find(
+        (item) => item.address === transaction.tokenAddress,
+      );
+      return token;
+    });
+  }, [values, colony.tokens]);
+
+  const handleTabToggle = useCallback(
+    (newIndex: number) => {
+      const newTransactionTabs = transactionTabStatus.map((tab, index) =>
+        index === newIndex ? !tab : tab,
+      );
+      setTransactionTabStatus([...newTransactionTabs]);
+    },
+    [transactionTabStatus, setTransactionTabStatus],
+  );
```

Comment:
What's the value in memoising this function? As far as I can tell, it gets re-run every render anyway because inside it you call `setTransactionTabStatus` and have `transactionTabStatus` as one of its dependencies.

Reply:
For handleTabToggle the idea was to only rerun this function when the user clicks to open/close a tab but I see your point. I was originally following the pattern I found in GnosisControlSafeForm.tsx as that component works very similar and contains the same function. Do you agree it makes sense to not have the useCallback there as well? Line 209 GnosisControlSafeForm.tsx

Reply:
I got it wrong in my first comment, it's not being re-run _every_ render. I meant to say that it seems to be run the same number of times with and without the usecallback.

And yes, would have the same question re the rationale for memoisation in other similar cases.

Reply:
@willm30 I made the changes and removed the useCallback function. I also, confirmed with Olga that in some cases like in this one, the useCallback is not necessary.